Added description new logic for generating data for templating. (#2859)
kyma-project/test-infra: development/tools/cmd/rendertemplates/README.md

 
 
- **Local ConfigSets** defined under the **localSets** parameter for each **to** key:

```yaml
render:
  - to: ../prow/jobs/test-infra/buildpack.yaml
    localSets:
      default:
        skip_report: "false"
        max_concurrency: "10"
        branches:
        - "^master$"
      presubmit:
        type_presubmit: "true"
        labels:
          preset-build-pr: "true"
      postsubmit:
        type_postsubmit: "true"
        cluster: "trusted-workload"
```

   ConfigSets defined in **localSets** have a scope limited to the generated file in which they are defined. Use **localSets** to hold data that is common within the generated file.

- **One-job ConfigSets** defined in the **jobConfig** key:

```yaml
jobConfigs:
  - repoName: "github.com/kyma-project/test-infra"
    jobs:
    - jobConfig:
        name: "pre-test-infra-bootstrap"
        run_if_changed: "^prow/images/bootstrap/"
        args:
        - "/home/prow/go/src/github.com/kyma-project/test-infra/prow/images/bootstrap"
```

> ConfigSets defined in **jobConfig** set data for one job. It should keep values specific for this job only.

Reviewer: ```suggestion
ConfigSets defined in **jobConfig** set data for one job. Use such ConfigSets to keep values specific for one job only.
```

> Render Templates builds the **Values** variable by merging ConfigSets from **globalSets** first. If the job inherits the default ConfigSet from **globalSets**, it's merged first and all the other ConfigSets from **globalSets** are merged after. Then, ConfigSets from **localSets** are merged. Again, if the job inherits the default ConfigSet from **localSets**, it's merged first and then all other ConfigSets from **localSets** are merged. ConfigSets other than default are merged in any order during **globalSets** and **localSets** phases. ConfigSets from **jobConfig** are merged as the last ones. Existing keys in the **Values** variable are overwritten by values from the merged ConfigSets.

Reviewer: ```suggestion
Render Templates builds the **Values** variable by merging ConfigSets from **globalSets** first. If the job inherits the default ConfigSet from **globalSets**, it's merged first and all the other ConfigSets from **globalSets** are merged after. Then, ConfigSets from **localSets** are merged. Again, if the job inherits the default ConfigSet from **localSets**, it's merged first and then all the other ConfigSets from **localSets** are merged. ConfigSets other than default are merged in any order during the **globalSets** and **localSets** phases. ConfigSets from **jobConfig** are merged as the last ones. Existing keys in the **Values** variable are overwritten by values from the merged ConfigSets.
```
